apps/meteor/client/views/marketplace/hooks/useAppsCountQuery.ts

```diff
@@ -0,0 +1,41 @@
+import { useEndpoint } from '@rocket.chat/ui-contexts';
+import { useQueryClient, useQuery } from '@tanstack/react-query';
+import { useCallback } from 'react';
+
+type Variant = 'success' | 'warning' | 'danger';
+
+const getProgressBarValues = (numberOfEnabledApps: number, enabledAppsLimit: number): { variant: Variant; percentage: number } => ({
+	variant: 'success',
+	...(numberOfEnabledApps + 1 === enabledAppsLimit && { variant: 'warning' }),
+	...(numberOfEnabledApps >= enabledAppsLimit && { variant: 'danger' }),
+	percentage: Math.round((numberOfEnabledApps / enabledAppsLimit) * 100),
+});
+
+export const useAppsCountQuery = (context: 'private' | 'explore' | 'marketplace') => {
+	const getAppsCount = useEndpoint('GET', '/apps/count');
+
+	return useQuery(
+		['apps/count', { context }],
```

Comment:
Curious, why contxt needs to be an object?

Reply:
@felipe-rod123

Reply:
`context` is not an object, it's a string. Sending an object as a second element for `queryKey` turns it into a parameter - so `useQuery` will memoize/cache results depending on `context`. If we remove it, `useQuery` will use the same results for all contexts.